### PR TITLE
[open62541] add amalgamation feature

### PR DESCRIPTION
**Pull request by Neumann-A**
required by qt6

### CODE REVIEW

File: versions/baseline.json [hunk 1]
```diff
@@ -2398,7 +2398,7 @@
     },
     "harfbuzz": {
       "baseline": "2.7.4",
-      "port-version": 1
+      "port-version": 2
```
Comment (Neumann-A) [inline versions/baseline.json:2401]:
```suggestion
      "port-version": 1
```

File: ports/open62541/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "open62541",
+  "version-string": "1.1.2",
```
Comment (NancyLi1013) [inline ports/open62541/vcpkg.json:3]:
```suggestion
  "version": "1.1.2",
```
This field might be `version `here. Please see https://github.com/microsoft/vcpkg/blob/master/docs/specifications/versioning.md#22-package-versions

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @Neumann-A 

Thanks for your PR.
Have you tested this new feature `amalgamation`?

**Comment by Neumann-A** [comment]
@NancyLi1013 the feature is used in #14333. One of the qt6 modules requires it.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @Neumann-A.

**Pull request merged**